sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/WriteToDataSourceV2.scala

```diff
@@ -110,7 +108,7 @@ object DataWritingSparkTask extends Logging {
       useCommitCoordinator: Boolean): WriterCommitMessage = {
     val stageId = context.stageId()
     val partId = context.partitionId()
-    val attemptId = context.attemptNumber()
+    val attemptId = context.taskAttemptId().toInt
```

Comment:
nit: can you rename this variable `tid`?  these names are pretty confusing, but I think that at least "tid" is used consistently and exclusively for this, while "attempt" means a lot of different things.

Reply:
I was going to suggest removing the cast to int, but well, that's in `DataWriterFactory` and would be an API breakage... hopefully won't cause issues aside from weird output names when the value overflows the int.

Reply:
HadoopWriteConfigUtil has the same issue, its a public interface and uses in for attempt number. 
it seems somewhat unlikely but more likely to be able to go over an int for task ids in spark then in say MapReduce.  we do have partitionId as an Int so if partitions go to Int and you have task failures then taskids could go over Int.  Looking at our options